src/libraries/System.ComponentModel.Annotations/src/System/ComponentModel/DataAnnotations/ValidationAttribute.cs

```diff
@@ -330,7 +331,10 @@ public virtual bool IsValid(object value)
             }
 
             // call overridden method.
-            return IsValid(value, null) == ValidationResult.Success;
+            // TODO-NULLABLE: The null here seems to be a bug - current sub-types overriding IsValid(value, validationContext) require a value here,
+            // and throw NRE. This entry point is probably never used - Validator goes through GetValidationResult which always passes
```

Comment:
Yeah, we guaranteed that this code path could never get hit in that way when validated through the APIs. One could construct the attributes and call the methods directly to reach it, but that would be closed as by design.

There's no validation context that could be conjured up, so this design set it up to fail with an exception rather than failing by constructing an empty validation context that could then have bad consequences including successful validation when that might not really be true.

For some backstory on the awkwardness around these methods, I happened to write a blog post back in the day: https://jeffhandley.com/2014-04-15/a-fun-validationattribute-bug

It might be worth adding more comments in here though.  Maybe something like:

``` csharp
// The IsValid method without a validationContext predates the one accepting the context.
// This is theoretically unreachable through normal use cases.
// Instead, the overload using validationContext should be called.
```

Reply:
Nice to have reviewing from the person who was originally involved in this stuff :smile: Will change to your comment of course.

Reply:
Thanks. I just wish I had left better code comments in here originally since this is so convoluted and misleading.